app/ts/components/Submit.tsx

```diff
@@ -63,171 +72,175 @@ const SimulationResult = ({
 		if (missingSigners) return 'Missing private keys for signing accounts.'
 		if (insufficientBalance) return 'Funding wallet has insufficent balance.'
 		return false
 	})
 
-	async function ensureRelayProvider() {
-		const relay = appSettings.peek().relayEndpoint
-		return flashbotsProvider.value && flashbotsProvider.value.connection.url === relay ? flashbotsProvider.value : await createProvider(provider, relay)
+	async function simulateCallback() {
+		if (!provider.value) throw 'User not connected'
+		if (!bundle.value) throw 'No imported bundle found'
+		const simulationResult = await simulateBundle(
+			bundle.value,
+			fundingAmountMin.peek(),
+			provider.value,
+			signers.peek(),
+			blockInfo.peek(),
+			appSettings.peek()
+		)
+		if ('error' in simulationResult) throw new Error((simulationResult as RelayResponseError).error.message)
+		else return simulationResult
 	}
 
-	async function simulateBundle() {
-		try {
-			const relayProvider = await ensureRelayProvider()
-			if (!flashbotsProvider.value) flashbotsProvider.value = relayProvider
-			if (!provider.value) throw 'User not connected'
-			if (!interceptorPayload.value) throw 'No imported bundle found'
-			const simulationResult = await simulate(
-				relayProvider,
-				provider.value.provider,
-				blockInfo.peek(),
-				appSettings.peek().blocksInFuture,
-				interceptorPayload.value,
-				signers.peek(),
-				fundingAmountMin.peek(),
-			)
-			if ('error' in simulationResult) throw new Error((simulationResult as RelayResponseError).error.message)
-			else return simulationResult
-		} catch (error) {
-			console.error({ error })
-			throw error
-		}
-
-	}
+	const outstandingBundles = useSignal<PendingBundle>({ bundles: {} })
 
 	async function bundleSubmission(blockNumber: bigint) {
-		const relayProvider = await ensureRelayProvider()
-		if (!flashbotsProvider.value) flashbotsProvider.value = relayProvider
-		if (!provider.value) throw 'User not connected'
-		if (!interceptorPayload.value) throw 'No imported bundle found'
+		submissionStatus.value = { ...submissionStatus.peek(), lastBlock: blockNumber }
 
-		const bundleSubmission = await sendBundle(
-			relayProvider,
-			provider.value.provider,
-			{ ...blockInfo.peek(), blockNumber },
-			appSettings.peek().blocksInFuture,
-			interceptorPayload.value,
-			signers.peek(),
-			fundingAmountMin.peek(),
-		).catch(() => {
-			bundleStatus.value = {
-				active: bundleStatus.value.active,
-				lastBlock: blockNumber,
-				pendingBundles: [...bundleStatus.value.pendingBundles, { hash: '', state: 'rejected', details: `RPC Error: ${blockNumber.toString()}` }],
-			}
-		})
+		if (!provider.value) throw new Error('User not connected')
+		if (!bundle.value) throw new Error('No imported bundle found')
+
+		// Check status of current bundles
+		// @DEV: Checked provider.value above, but LSP thinks it can still be undefined, so we cast it
```

Comment:
this is because of await, you can store it in const variable and then use that const variable